lxd/instance_post.go

```diff
@@ -716,3 +686,31 @@ func instancePostCreateContainerMountPoint(d *Daemon, project, containerName str
 
 	return nil
 }
+
+func migrateInstance(d *Daemon, r *http.Request, inst instance.Instance, projectName string, targetNode string, sourceNodeOffline bool, name string, instanceType instancetype.Type, req api.InstancePost) (func(op *operations.Operation) error, error) {
```

Comment:
This looks like a pretty odd signature. Any reason why we can't make `migrateInstance` just:

```
 func migrateInstance(d *Daemon, r *http.Request, inst instance.Instance, projectName string, targetNode string, sourceNodeOffline bool, name string, instanceType instancetype.Type, req api.InstancePost, op *operations.Operation) error {
```

Then you can wrap that into a Run function when called from `instancePost` and can also call it directly from the evacuation code.